backend/game_flow/game_flow_websocket.py

```diff
@@ -75,25 +83,59 @@ def end_game(game_id):
+        }))
+
+        lobby: Optional[Lobby] = lobby_repository.get_lobby(game_id)
+
+        if lobby and player_id in lobby['players']:
+            del lobby['players'][player_id] 
+
+        if not lobby['players']:
+            lobby_repository.remove_lobby(game_id)
+
+
 def socket_handler(websocket):
-    while True:
-        message = websocket.receive()
-        websocket_message: WebsocketMessage = websocket_message_from_dict(json.loads(message))
-        if websocket_message.method == 'create':
-            create_game(websocket_message.game_id, websocket_message.player_id, websocket)
-        elif websocket_message.method == 'join':
-            join_game(websocket_message.player_id, websocket_message.game_id, websocket)
-        elif websocket_message.method == 'init':
-            init_game(websocket_message.player_id, websocket_message.game_id)
-        elif websocket_message.method == 'score':
-            try:
-                player_move_req = player_move_from_dict(websocket_message.payload)
-            except:
-                websocket.send(PlayerScoredResponse(0, [], "bad request"))
-            else:
-                player_move = PlayerMove(websocket_message.player_id, player_move_req.symbol_id,
-                                    player_move_req.middle_card_id)
-                handle_score(websocket, player_move, websocket_message.game_id)
+    player_id = None
+    game_id = None
+
+    try:
+        while True:
+            message = websocket.receive()
+            websocket_message: WebsocketMessage = websocket_message_from_dict(json.loads(message))
+
+            if websocket_message.method == 'create':
+                create_game(websocket_message.game_id, websocket_message.player_id, websocket)
+                player_id = websocket_message.player_id
+                game_id = websocket_message.game_id
+            elif websocket_message.method == 'join':
+                join_game(websocket_message.player_id, websocket_message.payload['player_name'], websocket_message.game_id, websocket)
+                player_id = websocket_message.player_id
+                game_id = websocket_message.game_id
+            elif websocket_message.method == 'init':
+                init_game(websocket_message.player_id, websocket_message.game_id)
+            elif websocket_message.method == 'score':
+                try:
+                    player_move_req = player_move_from_dict(websocket_message.payload)
+                    player_move = PlayerMove(websocket_message.player_id, player_move_req.symbol_id, player_move_req.middle_card_id)
+                    handle_score(websocket, player_move, websocket_message.game_id)
+                except Exception as e:
+                    websocket.send(PlayerScoredResponse(0, [], "bad request"))
+
+    except Exception as e:
+        print(f"WebSocket error: {e}")
+
+    finally:
+        if player_id and game_id:
+            remove_player(game_id, player_id)
```

Comment:
if we reach this point it will mean that the ws exploded right? shouldn't we just close/deregister all the connections/players/games here?

Reply:
You can see an example of what I mentioned in the video demo